main.nf

```diff
@@ -11,60 +11,133 @@
-
-def makeBaseName = { f -> """\
-${f.getSimpleName()}.pggb-\
-s${params.edyeet_segment_length}-\
-p${params.edyeet_map_pct_id}-\
-n${params.edyeet_n_secondary}-\
-a${params.edyeet_align_pct_id}-\
-K${params.edyeet_mash_kmer}\
-${edyeet_merge_cmd}\
-${edyeet_split_cmd}\
-${edyeet_exclude_cmd}-\
+def alignment_merge_cmd = params.alignment_merge_segments ? "-M" : params.alignment_merge_cmd
+def alignment_exclude_cmd = params.alignment_exclude_delim ? "-Y${params.alignment_exclude_delim}" : params.alignment_exclude_cmd
+def alignment_split_cmd = params.alignment_no_splits ? "-N" : params.alignment_split_cmd
+def aligner = params.wfmash ? "W" : "E"
+def edyeet_align_pct_id_display = params.wfmash ? "" : "a${params.edyeet_align_pct_id}-"
+def smoothxg_poa_params_display = params.smoothxg_poa_params.replaceAll(/,/, "_")
+def file_name_prefix_display = ""
+def alignment_prefix = ""
+def seqwish_prefix = ""
+def smoothxg_prefix = ""
+// default case
+if (!params.file_name_prefix) {
+  file_name_prefix_display = ".pggb"
+  alignment_prefix = "-${aligner}"
+  seqwish_prefix = ".seqwish"
+  smoothxg_prefix = ".smoothxg"
+} else if (params.file_name_prefix == "pggb") {
+  // fancy naming scheme
+  file_name_prefix_display = ".pggb"
+  alignment_prefix = """-\
+${aligner}-\
+s${params.alignment_segment_length}-\
+l${params.alignment_block_length}-\
+p${params.alignment_map_pct_id}-\
+n${params.alignment_n_secondary}-\
+${edyeet_align_pct_id_display}\
+K${params.alignment_mash_kmer}\
+${alignment_merge_cmd}\
+${alignment_split_cmd}\
+${alignment_exclude_cmd}\
+"""
+  seqwish_prefix = """\
+.seqwish-\
 k${params.seqwish_min_match_length}-\
+B${params.seqwish_transclose_batch}\
+"""
+  smoothxg_prefix = """${seqwish_prefix}\
+.smoothxg-\
 w${params.smoothxg_max_block_weight}-\
 j${params.smoothxg_max_path_jump}-\
-W${params.smoothxg_min_subpath}-\
 e${params.smoothxg_max_edge_jump}-\
-I${params.smoothxg_block_id_min}\
+I${params.smoothxg_block_id_min}-\
+p${smoothxg_poa_params_display}-M-J0.7-K-G150\
+"""
+} else {
+  // take the given prefix
+  file_name_prefix_display= "${params.file_name_prefix}.pggb"
+  aligment_prefix = "-${aligner}"
+  seqwish_prefix = ".seqwish"
+  smoothxg_prefix = ".smoothxg"
+}
+
+// TODO update, make_file_prefix
+def make_file_prefix = { f -> """\
+${f.getName()}${file_name_prefix_display}\
 """ }
 
-fasta = channel.fromPath("${params.input}").map { f -> tuple(makeBaseName(f), f) }
+def make_file_prefix_given = { f -> """\
+${file_name_prefix_display}\
+""" }
+
+if (!params.file_name_prefix || params.file_name_prefix == "pggb") {
+  fasta = channel.fromPath("${params.input}").map { f -> tuple(make_file_prefix(f), f) }
+} else {
+  fasta = channel.fromPath("${params.input}").map { f -> tuple(make_file_prefix_given(f), f) }
+}
 
 process edyeet {
   input:
   tuple val(f), path(fasta)
 
   output:
-  tuple val(f), path(fasta), path("${f}.paf")
+  tuple val(f), path(fasta), path("${f}${alignment_prefix}.paf")
```

Comment:
Still not a fan of writing this into the file name. But we argued about this at length already :)

Reply:
As suggested @https://github.com/nf-core/pangenome/pull/44#discussion_r583440561, the default behavior is to have really short names of the files.

Reply:
What do @heuermh @andreaguarracino think?

Reply:
A marriage is made up of compromises. I hope this is a good one.